### DISCUSSION TITLE
Assigning ID on material

### DISCUSSION THREAD

**Discussion by Jina-Kim-1108** [jspsych/jsPsych#958] [Q&A] [answered]
Hi! Thanks so much for your reply in advance!
I have a question for assigning ID on each trial.
Below is the code of my study. Since I randomize the order of ‘text’ here, I would like to assign ID from I to 4 on each text so that I can sort all participants data accordingly.
For cloze plug-in, the data is answers. Can I also get ‘check_answers’ value for each trial?
Here is the code.

```js
var survey = {
	type: 'cloze',
	text: jsPsych.timelineVariable('text'),
	check_answers: true,
	button_text: 'Next'
};

var actual_survey = {
	timeline: [survey],
	timeline_variables: [
       		{text: "In the rain forest," + "<p>lion is only % a %."},
       		{text: "In the river," + "<p>I saw fish eats % b %."},
       		{text: "Practice 3," + "<p>Praceice% b %."},
       		{text: "Practice 4," + "<p>Praceice% b %."},
       		{text: "Practice 5," + "<p>Praceice% b %."}
       	],
       	randomize_order : true
}
```

**Comment by jodeleeuw** [comment, marked as answer]
You can tag trials with additional data through the `data` parameter. You can combine this with timeline variables.

```js
var survey = {
	type: 'cloze',
	text: jsPsych.timelineVariable('text'),
	check_answers: true,
	button_text: 'Next',
	data: {
		trial_id: jsPsych.timelineVariable('id')
	}
};

var actual_survey = {
	timeline: [survey],
	timeline_variables: [
       		{text: "In the rain forest," + "<p>lion is only % a %.", id: 1},
       		{text: "In the river," + "<p>I saw fish eats % b %.", id: 2},
       		{text: "Practice 3," + "<p>Praceice% b %.", id: 3},
       		{text: "Practice 4," + "<p>Praceice% b %.", id: 4},
       		{text: "Practice 5," + "<p>Praceice% b %.", id: 5}
       	],
       	randomize_order : true
}
```

> For cloze plug-in, the data is answers. Can I also get ‘check_answers’ value for each trial?

Do you want whether the subject answered correctly in the data? I believe in the cloze plugin if the subject answers incorrectly the trial doesn't end. Instead, the `mistake_fn` is called. If you want to allow incorrect responses then you may have to modify the plugin slightly.

**Reply by Jina-Kim-1108** [reply]
Hi! Thanks so much for your reply! and Yes, I already modified the plug-in to allow participant's incorrect answers. What I was trying to do is to have the correct answer in the  output data. current output only shows my answer like below. So is there any code to put correct answer in the output data?

"answers": [
			"x"
		],
		"trial_type": "cloze",
		"trial_index": 20,
		"time_elapsed": 28005,
		"internal_node_id": "0.0-5.0-8.0-0.4",
		"subject": "5t2zz"

**Reply by jodeleeuw** [reply]
You could add an additional property here: 

https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/plugins/jspsych-cloze.js#L93-L95

I believe that the variable `solutions` has the correct answers. So you could add `correct_answers: solutions` to the `trial_data` object.

**Reply by Jina-Kim-1108** [reply]
Thank you so much for the code! It worked perfectly.
I have one other questions for you here. 
I added 'ID' next to 'text', but still I cannot see the text ID on output data. Could you tell me why it does not show up my output?

var actual_survey ={
                    timeline: [survey],
                    timeline_variables: [
                {text: "In the rain forest," + "<p>lion is only % a %.", id: 1},
                {text: "In the river," + "<p>I saw fish eats % b %.", id: 2},
                {text: "Practice 3," + "<p>Praceice% b %.", id: 3},
                {text: "Practice 4," + "<p>Praceice% b %.", id: 4},
                {text: "Practice 5," + "<p>Praceice% b %.", id: 5}
                ],
                randomize_order : true,
                data: jsPsych.timelineVariable('text')
                }

How can I bring text id on my output data?
I guess I can use 'jsPsych.data.addProperties({id : text_id})' but I have no clue how to define 'text_id' here.

**Reply by jodeleeuw** [reply]
Great! Glad that worked.

Did you add the `data` property to the `survey` object? See the first part of my code example above. 

[Part 9 of the tutorial](https://www.jspsych.org/tutorials/rt-task/#part-9-tagging-trials-with-additional-data) explains how to add data to trials. It might be helpful to read that over.

**Reply by Jina-Kim-1108** [reply]
Thanks a lot for the help! I really appreciate it! :)
